pkg/common/ipsec/utils.go

```diff
@@ -15,16 +17,22 @@ const (
 	markStateOut = 0xe00
 )
 
-func CountUniqueIPsecKeys(states []netlink.XfrmState) int {
+func CountUniqueIPsecKeys(states []netlink.XfrmState) (int, error) {
 	keys := make(map[string]bool)
 	for _, s := range states {
-		if s.Aead == nil {
+		if s.Aead != nil && s.Auth == nil && s.Crypt == nil {
+			keys[string(s.Aead.Key)] = true
+			continue
+		}
+		if s.Aead == nil && s.Auth != nil && s.Crypt != nil {
+			// we want to count the number of unique (Auth, Crypt) tuples
+			key := fmt.Sprintf("%s:%s", string(s.Auth.Key), string(s.Crypt.Key))
+			keys[key] = true
 			continue
 		}
-		keys[string(s.Aead.Key)] = true
+		return 0, fmt.Errorf("an unsupported XfrmStateAlgo combination has been found")
```

Comment:
Fixed.

Reply:
Then I don't think it makes sense to return the error at all.
What do you think if I revert it to the initial function signature and just print to StdErr in the loop in case of an invalid combination?

Reply:
Done.
Does it make sense to squash the commits for this PR? It looks pretty small.
CC: @pchaigno

Reply:
Done.